src/CoreWCF.Primitives/src/CoreWCF/Channels/ByteStreamMessageEncoder.cs

```diff
@@ -79,7 +75,11 @@ public override Message ReadMessage(ArraySegment<byte> buffer, BufferManager buf
             //    TD.ByteStreamMessageDecodingStart();
             //}
 
-            var messageData = new ByteStreamBufferedMessageData(buffer, bufferManager);
+            int bufferLength = (int)buffer.Length;
+            byte[] bytes = bufferManager.TakeBuffer(bufferLength);
+            buffer.CopyTo(bytes.AsSpan(0, bufferLength));
+
```

Comment:
This is still extra allocation. ByteSTreamBufferedMessageData needs to be converted to use a ReadOnlySequence on its constructor.

Reply:
done